[WORKFLOW/ETHEREUM-CONTRACTS] Fix Up Coverage

0xdavinchee opened the pull request:
## Summary
The ethereum-contracts test coverage has "dropped" to around 85% on codecov, however, this is largely due to several factors:
- new test files written specifically for foundry 
- new foundry test code coverage results not being merged and uploaded to codecov
- removal of previous test files
- commented out `lcov --add-tracefile` in `ci.canary`.

## Changes
- Ignore test contracts and other contracts as part of coverage
- Add some tests for BatchLiquidator/SuperAppBaseCFA foundry
- Recover CFAv1Library and IDAv1Library tests
- Add missing SuperTokenV1Library tests (w/ userData)
- Add missing SuperfluidGovernanceII test (setPPPConfig with invalid liq/pat period)
- extract more foundry specific tests
- fix typo in SuperToken.sol
- _revertWhenOutOfGas can and should be a view function
  - Add `_revertWhenOutOfGas` prop test

0xdavinchee committed:
fix up coverage tests

- Add some tests to be pedantic for BatchLiquidator/SuperAppBaseCFA foundry
- Recover CFAv1Library and IDAv1Library tests
- fix typo in SuperToken.sol
- extract more foundry specific tests

hellwolf commented:
```
codecov:
  require_ci_to_pass: yes
```

Does this prevent us from having the result if canary is red (which seems a common case atm) ?

hellwolf commented on packages/ethereum-contracts/.solcover.js:
```diff
@@ -5,6 +5,11 @@ module.exports = {
     skipFiles: [
         "mocks/",
         "apps/SuperAppBase.sol",
+        "utils/SuperfluidFrameworkDeployer",
```
Is SuperfluidFrameworkDeployer really a good candidate to skip?

Same goes for SuperTokenDeployer

0xdavinchee replied:
It depends, we don't have an explicit definition of what files should be included in our coverage.
In my opinion, we should reduce noise as much as possible and it should solely be the core contracts which are deployed in production.
These are both peripheral contracts that are not used in production and we do already have tests for them.

hellwolf replied:
I thought they are used in some of the test cases, hence they should have been covered. I guess my assumption was wrong?

hellwolf commented on packages/ethereum-contracts/.solcover.js:
```diff
@@ -5,6 +5,11 @@ module.exports = {
     skipFiles: [
         "mocks/",
         "apps/SuperAppBase.sol",
```
what's the role of SuperAppBase nowadays

0xdavinchee replied:
I believe it is used by the `SuperAppBaseCFA` and it's usually overridden, we could create two simple SuperApp's  which inherits from the respective bases and override the functions, but just use super.function, although not sure if what we are testing here is interesting.

0xdavinchee commented:
> ```
> codecov:
>   require_ci_to_pass: yes
> ```
> 
> Does this prevent us from having the result if canary is red (which seems a common case atm) ?

Yeah, I believe so.

0xdavinchee committed:
ci.feature codecov

0xdavinchee committed:
remove extraction step

hellwolf commented:
installed codecov to github for the org.

0xdavinchee committed:
fix up coverage

- add tests for supertokenlibv1 coverage
- rename idaV1Lib => superTokenLibIDA and cfaV1Lib => superTokenLibCFA

0xdavinchee committed:
add comments for reason for skipping coverage

0xdavinchee committed:
cleanup

- add back extraction
- move ignore list to codecov.yml
- remove skipFiles from solcover.js
- add test for SF_GOV_INVALID_LIQUIDATION_OR_PATRICIAN_PERIOD
- add safegas prop test
- try out separate upload to allow codecov to merge on their end

0xdavinchee committed:
>= 63

the multiplication takes some gas

0xdavinchee committed:
bring back extract n merge

0xdavinchee committed:
it likes the skipFiles

no extract step + add back skipFiles

0xdavinchee committed:
extraction required

0xdavinchee committed:
safe gas

0xdavinchee committed:
out of gas consistency

hellwolf commented on packages/ethereum-contracts/test/foundry/libs/SafeGasLibrary.prop.t.sol:
```diff
@@ -0,0 +1,24 @@
+/// We mock this here by using a multiple of the gasleft() before the function call.
+contract SafeGasLibraryProperties is Test {
+
+    function test_SafeGasRevertsAsExpected(uint16 multiple) public {
```
This is a bad test, you are testing exactly how it is coded so you are not actually testing anything useful.

But what you really want to test is the property that: when you give an arbitrary gas limit to a call and if it reverts OUT_OF_GAS it is less than 63*.

0xdavinchee replied:
removed 😅

0xdavinchee committed:
remove safe gas lib test

0xdavinchee committed:
Merge branch 'dev' into coverage-fix

0xdavinchee committed:
cleanup extraction!

hellwolf reviewed (approved):
Thank you! This is great. Back to 98%+

0xdavinchee committed:
Revert "cleanup extraction!"

This reverts commit 861240c1853aff893ade45742b1259f1991653ac.

0xdavinchee committed:
cleaner extraction

- extract out only packages/ethereum-contracts/contracts for coverage
- remove files we don't care about for coverage (see .solcover.js)

0xdavinchee committed:
remove mocks
